crates/task-impls/src/quorum_vote.rs

```diff
@@ -70,12 +74,170 @@ struct VoteDependencyHandle<TYPES: NodeType, I: NodeImplementation<TYPES>> {
     view_number: TYPES::Time,
     /// Event sender.
     sender: Sender<Arc<HotShotEvent<TYPES>>>,
+    /// The current version of HotShot
+    version: Version,
+}
+
+impl<TYPES: NodeType, I: NodeImplementation<TYPES> + 'static> VoteDependencyHandle<TYPES, I> {
+    /// Updates the shared consensus state with the new voting data.
+    async fn update_shared_state(
+        &self,
+        proposal: &QuorumProposal<TYPES>,
+        proposed_leaf: &Leaf<TYPES>,
```

Comment:
This seem to basically be duplicates of the same data, can we get away with just passing the leaf?

Reply:
Yep